scripts/sanitycheck

```diff
@@ -1661,9 +1667,13 @@ class TestInstance:
                 content = "\n".join(self.testcase.extra_configs)
 
             if options.enable_coverage:
-                if platform in options.coverage_platform:
+                if platform.name in options.coverage_platform:
                     content = content + "\nCONFIG_COVERAGE=y"
 
+            if not options.disable_asan:
+                if platform.type == "native":
+                    content = content + "\nCONFIG_ASAN=y"
```

Comment:
could we select this in Kconfig instead?

historically, having sanitycheck build differently than local builds of the same code has led to all kinds of confusion.

Reply:
Yes this would be possible, but it would mean that ASAN is enabled by default for posix arch boards and require libasan to be installed.

Further I did observe deadlocks* while running sanitycheck with ASAN memory leak detection enabled, which is by default enabled, and it needs to be disabled by setting an environment variable. Enabling ASAN by default would mean that all user of the posix arch should be aware of this deadlock possibility and that they need to set an environment variable.

* Until now is I was not able to pinpoint the root cause of the deadlock but it points to a race condition between LSAN (memory leak detector in ASAN) running its memory leak detection and the the shutdown of the posix arch spawned threads.

Reply:
> Yes this would be possible, but it would mean that ASAN is enabled by default for posix arch boards and require libasan to be installed

Right, but since we're doing this in sanitycheck I think we have a libasan dependency already.

> (1) Until now is I was not able to pinpoint the root cause of the deadlock but it points to a race condition between LSAN (memory leak detector in ASAN) running its memory leak detection and the the shutdown of the posix arch spawned threads.

If there isn't a GH issue for this already please file one.

@aescolar any thoughts on this?

Reply:
@nashif @aescolar Any input on enabling ASAN by default in Kconfig instead of enabling it only by default in sanitycheck?

@andrewboie 
> does native posix work in Windows?

No it does not work in windows, I dont know even how well native_posix is supported on mac

Reply:
> Any input on enabling ASAN by default 

Sorry but that is a no go. This discussion already happened 1.5 years ago. Just like we do not run by default posix arch boards in valgrind. Compiling with ASAN is nice for native_posix in many cases, but is a no go in others. Even if everything else worked fine, it adds a lot of runtime overhead, which for long BLE simulations is a dealbreaker.

I certainly understand @andrewboie 's point, and I think it is valid. But then the solution should be to have this sanitycheck option inverted (to `--enable-asan`). Then have `run_ci.sh ` call it so it is utilized in CI, but not by default in local runs (if @andrewboie thinks it would be sensible, otherwise just have it as an option for people to run locally just as the valgrind one)

> does native posix work in Windows?

Nope, Windows in a not a POSIX compliant OS :)

> I dont know even how well native_posix is supported on mac

It is known to not work (see 10945)

Reply:
@andrewboie Issue created for deadlock: #20122